src/osu_allgather.jl

```diff
@@ -0,0 +1,38 @@
+export OSUAllgather
+
+struct OSUAllgather <: MPIBenchmark
+    conf::Configuration
+    name::String
+end
+
+function OSUAllgather(T::Type=Float32;
+                   filename::Union{String,Nothing}="julia_osu_allgather.csv",
+                   kwargs...,
+                   )
+    return OSUAllgather(
+        Configuration(T; filename, max_size=2 ^ 20, kwargs...),
+        "OSU Gather",
+    )
+end
+
+function osu_allgather(T::Type, bufsize::Int, iters::Int, comm::MPI.Comm)
+    rank = MPI.Comm_rank(comm)
+    nranks = MPI.Comm_size(comm)
+    send_buffer = ones(T, bufsize)
+    recv_buffer = zeros(T, bufsize * nranks)
+    root = 0
+    timer = 0.0
+
+    MPI.Barrier(comm)
+    for _ in 1:iters
+        tic = MPI.Wtime()
+        MPI.Allgather!(send_buffer, UBuffer(recv_buffer, Cint(bufsize), Cint(nranks), MPI.Datatype(T)), comm)
```

Comment:
I believe you can also do
```suggestion
        MPI.Allgather!(send_buffer, recv_buffer, comm)
```

Reply:
After the modification it give the error "LoadError: ERROR: LoadError: DivideError: integer division error"

Reply:
Uhm, is it because size 0 isn't handled nicely?  This sounds like a possible bug in `MPI.jl`.